PM2021-009 - All services returning 404 in Kubernetes.md

```diff
@@ -0,0 +1,57 @@
+# All services returning 404 in Kubernetes
+
+## Summary
+Malformed IngressRoute manifest applied to Kubernetes triggered a bug in Traefik v2 causing no ingress traffic to be received nor routed. Due to the nature of the bug it first triggered after all the Traefik v2 pods got restarted.
+
+## Timeline
+
+All times in UTC.
+
+| Time             | Event                   |
+| :--------------- | ----------------------- |
+| 2021-09-01 11:34 | Malformed IngressRoute applied to Kubernetes |
+| 2021-09-06 07:30 | Automated Kuberentes node rollover started |
+| 2021-09-06 08:00 | All Kubernetes services returning 404 |
+| 2021-09-06 08:05 | Root cause investigation started |
+| 2021-09-06 08:10 | ALB targets unhealthy |
+| 2021-09-06 08:15 | ALB targets healthy after altering the healthcheck path |
+| 2021-09-06 08:20 | All Kubernetes services still returning 404, but now it is Traefik v2 and not the ALB's |
```

Comment:
I could be mis-remembering but wasn't it Traefik returning 404 the entire time? All of the nodes were marked as unhealthy and when that is the case the ALB still routes traffic (as opposed to if there were 1 or more healthy nodes, it would route to them only). Behaviour is described here - https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-troubleshooting.html#no-healthy-targets

Reply:
Thanks for highlighting the behaviour Richard.  I'd presumed (wrongly) that if there were no healthy targets then it wouldn't route traffic at all, but the link you posted confirms that when all targets go unhealthy it will still route traffic.  So based on that I think your comment is probably correct - the 404's were always coming from Traefik.